File: backend/src/main/java/com/crm/dto/BenutzerCreateDTO.java
package com.crm.dto;

import java.util.Set;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Size;

public record BenutzerCreateDTO(
        @NotBlank String benutzername,
        @Size(min = 8, message = "Passwort muss mindestens 8 Zeichen lang sein") String passwort,
        @NotBlank String vorname,
        @NotBlank String nachname,
        @NotBlank @Email String email,
        @NotEmpty Set<String> rollen,
        boolean aktiv
) {}

File: ciam/src/main/kotlin/com/crm/ciam/dto/BenutzerCreateDTO.kt
package com.crm.ciam.dto

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.Size

data class BenutzerCreateDTO(
    @field:NotBlank val benutzername: String,
    @field:Size(min = 8, message = "Passwort muss mindestens 8 Zeichen lang sein") val passwort: String?,
    @field:NotBlank val vorname: String,
    @field:NotBlank val nachname: String,
    @field:NotBlank @field:Email val email: String,
    @field:NotEmpty val rollen: Set<String>,
    val aktiv: Boolean
)
